tests/report_tests/plugin_tests/openstack/openstack.py

```diff
@@ -0,0 +1,162 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+import json
+
+from sos_tests import StageTwoReportTest, ubuntu_only
+from sos.utilities import shell_out
+
+from avocado.utils import distro
+from avocado.utils.software_manager import distro_packages
+
+
+class OpenstackConfScrubbedTest(StageTwoReportTest):
+    """Ensure that the various openstack confs are picked up and properly
+    scrubbed
+
+    :avocado: tags=stagetwo
+    """
+
+    sos_cmd = ('-o openstack_cinder,openstack_keystone,openstack_glance,'
+               'openstack_neutron,openstack_nova')
+    files = [
+        ('cinder.conf', '/etc/cinder/cinder.conf'),
+        ('keystone.domain1.conf',
+         '/etc/keystone/domains/keystone.domain1.conf'),
+        ('keystone.conf', '/etc/keystone/keystone.conf'),
+        ('glance-api.conf', '/etc/glance/glance-api.conf'),
+        ('neutron.conf', '/etc/neutron/neutron.conf'),
+        ('nova.conf', '/etc/nova/nova.conf'),
+    ]
+
+    packages = {
+        'rhel': ['openstack-cinder', 'openstack-glance',
+                 'openstack-heat-common', 'openstack-keystone',
+                 'openstack-nova'],
+        'Ubuntu': ['cinder-common', 'glance-common', 'heat-common',
+                   'keystone-common',  'nova-common']
+    }
+
+    openstack_only = True
+
+    osp_release = ""
+
+    def _prep_crb_repo(self):
+        if self.osp_release != "":
+            this_distro = distro.detect()
+            if this_distro.version == "8":
+                pkg = {this_distro.name: ["dnf-plugins-core"]}
+                installed = distro_packages.install_distro_packages(pkg)
+                if not installed:
+                    raise Exception(
+                        "Unable to install requested packages "
+                        "dnf-plugins-core")
+            shell_out("dnf config-manager --enable crb")
+
+    def setup_mocked_packages(self):
+        this_distro = distro.detect()
+        if this_distro.name == "centos" or this_distro.name == "centos-stream":
+            if this_distro.version == "9":
+                self.osp_release = "yoga"
+            if this_distro.version == "8":
+                self.osp_release = "victoria"
```

Comment:
Knowing OSP a little only, but why do we test on unmaintained releases? (https://releases.openstack.org/)

Reply:
Thanks for the link, just made it consistent to be yoga, and enable the right repos for 8/9 and got it through, I think